src/Microsoft.VisualStudio.ProjectSystem.Managed/PooledObjects/ObjectPool.cs

```diff
@@ -0,0 +1,145 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.Threading;
+
+namespace Microsoft.VisualStudio.ProjectSystem.Managed.PooledObjects
```

Comment:
We don't use "Managed" in a namespace. For this, I'd copy the CoreFx naming; "VisualStudio.Buffers".